dbms/src/Operators/AggregateConvergentSourceOp.cpp

```diff
@@ -31,6 +31,11 @@ AggregateConvergentSourceOp::AggregateConvergentSourceOp(
 
 OperatorStatus AggregateConvergentSourceOp::readImpl(Block & block)
 {
+    if unlikely (!agg_context->convertPendingDataToTwoLevel())
+    {
+        setNotifyFuture(agg_context);
+        return OperatorStatus::WAIT_FOR_NOTIFY;
+    }
```

Comment:
how about adding a new event to convert pending data to two level?

Reply:
Done.